test/makeLatexReport.m

```diff
@@ -0,0 +1,209 @@
+          print_verbatim_information(fh, 'matlab2tikz'    , stat.tikzStage.message);
+      end
+      fprintf(fh, '\n\\normalsize\n\n');
+  end
+
+  % now, finish off the file and close file and window
+  texfile_finish(fh, status);
+  fclose(fh);
+end
+% =========================================================================
+function texfile_init(texfile_handle)
+
+  fprintf(texfile_handle, ...
+           ['\\documentclass[landscape]{scrartcl}\n'                , ...
+            '\\pdfminorversion=6\n\n'                               , ...
+            '\\usepackage{amsmath} %% required for $\\text{xyz}$\n\n', ...
+            '\\usepackage{hyperref}\n'                              , ...
+            '\\usepackage{graphicx}\n'                              , ...
+            '\\usepackage{epstopdf}\n'                              , ...
+            '\\usepackage{tikz}\n'                                  , ...
+            '\\usetikzlibrary{plotmarks}\n\n'                       , ...
+            '\\usepackage{pgfplots}\n'                              , ...
+            '\\pgfplotsset{compat=newest}\n\n'                      , ...
+            '\\usepackage[margin=0.5in]{geometry}\n'                , ...
+            '\\newlength\\figurewidth\n'                            , ...
+            '\\setlength\\figurewidth{0.4\\textwidth}\n\n'          , ...
+            '\\begin{document}\n\n']);
+
+end
+% =========================================================================
+function texfile_finish(texfile_handle, status)
+
+  [env,versionString] = getEnvironment();
+  
+  testsuites = unique(cellfun(@(s) func2str(s.testsuite) , status, ...
+                     'UniformOutput', false));                 
+  testsuites = name2tex(strjoin(testsuites, ', '));
+
+  fprintf(texfile_handle, ...
+      [
+      '\\newpage\n',...
+      '\\begin{tabular}{ll}\n',...
+      '  Suite    & ' testsuites ' \\\\ \n', ...
+      '  Created  & ' datestr(now) ' \\\\ \n', ...
+      '  OS       & ' OSVersion ' \\\\ \n',...
+      '  ' env '  & ' versionString ' \\\\ \n', ...
+      VersionControlIdentifier, ...
+      '  TikZ     & \\expandafter\\csname ver@tikz.sty\\endcsname \\\\ \n',...
+      '  Pgfplots & \\expandafter\\csname ver@pgfplots.sty\\endcsname \\\\ \n',...
+      '\\end{tabular}\n',...
+      '\\end{document}']);
+
+end
+% =========================================================================
+function print_verbatim_information(texfile_handle, title, contents)
+    if ~isempty(contents)
+        fprintf(texfile_handle, ...
+                ['\\subsubsection*{%s}\n', ...
+                 '\\begin{verbatim}\n%s\\end{verbatim}\n'], ...
+                title, contents);
+    end
+end
+% =========================================================================
+function texfile_addtest(texfile_handle, status)
+  % Actually add the piece of LaTeX code that'll later be used to display
+  % the given test.
+
+  ref_error = status.plotStage.error;
```

Comment:
Here we could check for issues in the `status` structure to add a layer of testsuite robustness.
Test runs currently fails here ungracefully, if the test was skipped. Certainly it makes sense to fix this, where the test is actually skipped, but an additional safety check here would (in my opinion) not hurt.